Bakker/models/bakker_koeken.py

```diff
@@ -0,0 +1,233 @@
+            betaalde_verkopen = record.verkoop_ids.filtered(lambda v: v.status == 'betaald')
+            record.totaal_verkocht = sum(betaalde_verkopen.mapped('aantal'))
+            record.totaal_omzet = sum(betaalde_verkopen.mapped('totaal_bedrag'))
+    
+    @api.depends('verkoop_ids')
+    def _compute_verkoop_count(self):
+        for record in self:
+            record.verkoop_count = len(record.verkoop_ids)
+    
+    @api.onchange('prijs_koek', 'voorraad_koek')
+    def _onchange_prijs_koek(self):
+        self._compute_totaal_inventarisatie()
+    
+    @api.onchange('totaal_inventarisatie')
+    def _onchange_totaal_inventarisatie(self):
+        self._inverse_totaal_inventarisatie()
+    
+    @api.depends('prijs_koek', 'voorraad_koek')
+    def _compute_totaal_inventarisatie(self):
+        for record in self:
+            record.totaal_inventarisatie = record.prijs_koek * record.voorraad_koek
+    
+    def _inverse_totaal_inventarisatie(self):
+        for record in self:
+            if record.prijs_koek != 0:
+                record.voorraad_koek = record.totaal_inventarisatie / record.prijs_koek
+            else:
+                record.voorraad_koek = 0
+            
+    @api.constrains('voorraad_koek', 'prijs_koek')
+    def _check_non_negative(self):
+        for record in self:
+            if record.voorraad_koek < 0:
+                raise ValidationError("Voorraad van de koek kan niet negatief zijn.")
+            if record.prijs_koek < 0:
+                raise ValidationError("Prijs van de koek kan niet negatief zijn.")
+        
+    def action_voorraad_bijvullen(self):
+        """Vul voorraad bij met standaard hoeveelheid"""
+        for record in self:
+            record.voorraad_koek += 20
+            record.aankoopdatum_koek = fields.Date.today()
+            
+        return {
+            'type': 'ir.actions.client',
+            'tag': 'reload',
+            'params': {
+                'title': 'Voorraad Bijgevuld!',
+                'message': f'Voorraad van {self.name_koek} is bijgevuld met 20 stuks',
+                'type': 'success',
+            }
+        }
+    
+    def action_uitverkocht(self):
+        """Markeer als uitverkocht"""
+        for record in self:
+            record.voorraad_koek = 0
+        return True
+    
+    def action_verse_batch(self):
+        """Maak nieuwe verse batch - duplicate record"""
+        for record in self:
+            # Maak een kopie van het huidige record
+            new_batch = record.copy({
+                'voorraad_koek': 30,
+                'aankoopdatum_koek': fields.Date.today(),
+                'vervaldatum_koek': fields.Date.today() + timedelta(days=30),
+                'name_koek': f"{record.name_koek} - Verse Batch",
+            })
+            
+            vers_tag = self.env['bakker_koeken_tags'].search([('name', '=', 'Vers')], limit=1)
+            if vers_tag:
+                new_batch.tags_ids = [(4, vers_tag.id)]
```

Comment:
```suggestion
        vers_tag = self.env['bakker_koeken_tags'].search([('name', '=', 'Vers')], limit=1)
        for record in self:
            # Maak een kopie van het huidige record
            new_batch = record.copy({
                'voorraad_koek': 30,
                'aankoopdatum_koek': fields.Date.today(),
                'vervaldatum_koek': fields.Date.today() + timedelta(days=30),
                'name_koek': f"{record.name_koek} - Verse Batch",
            })
            
            if vers_tag:
                new_batch.tags_ids = [(4, vers_tag.id)]
```

As you are searching with a hardcoded 'Vers' the search can be outside the loop and called once only